routes/api/channels.js

```diff
@@ -0,0 +1,61 @@
+var keystone = require('keystone');
+var User = keystone.list('User');
+var rp = require('request-promise');
+var _ = require('underscore');
+
+exports.getAll = function (req, res) {
+	var options = {
+		uri: 'https://slack.com/api/channels.list?',
+		qs: {
+			token: 'xoxp-2186825377-14603377189-42848621732-2e85220052'
+		},
+		json: true
+	};
+
+	rp(options)
+		.then(function (result) {
+			var channels = [];
+			var channelPromises = [];
+			_.each(result.channels, function (channel) {
+				channelPromises.push(new Promise(function (parentResolve, parentReject) {
+
+					var channelObject = channel;
+
+					var memberPromiseArray = [];
+					var memberArray = [];
+					_.each(channel.members, function (member) {
+
+						memberPromiseArray.push(new Promise(function (resolve, reject) {
+							User.model.find({slack_id: member}).exec(function (err, user) {
```

Comment:
findOne({slack_id: member}) gives back an object so you don't have to check for user[0]